### ISSUE TITLE
inbound parse webhook incorrectly parses e-mail addresses including encoded words

### ISSUE THREAD

**Issue by yvesf** [sendgrid/docs#6289]
Scenario: we receive an e-mail from a user that felt really lucky that he could use an address like: `From: =?UTF-8?B?PG15LW5hbWUtYWdhaW5AaG9zdC5sb2NhbD4=?= <my-name@host.local>`.

The inbound parse webhook will send us this in the “from” field as the value: `<my-name-again@host.local> <isamaag@TEST>` which is not a legal e-mail address anymore.

The original one may be really strange and close to a bad idea, although it is legal as it conforms to RFCs. 
I think the relevant RFCs are 5322 and 2047.  RFC-2047 mentions: “IMPORTANT: 'encoded-word's are designed to be recognized as 'atom's”.

The address syntax is specified in 5322, in this case it can be applied like this:

`address         =   mailbox / group` dropping group, which in our case expands to:
`address         =   (name-addr / addr-spec)` dropping add-spec, expands to:
`address         =   [display-name] angle-addr` expanding display-name, leads to:
`address         =   phrase angle-addr` expanding phrase leads to 
`address         =   (1*word / obs-phrase) angle-addr` In our case it ends as
`address         =   atom angle-addr` which is equal to:
`address         = =?UTF-8?B?PG15LW5hbWUtYWdhaW5AaG9zdC5sb2NhbD4=?= <my-name@host.local>`

This proves that the address is sound, you can have `<foo@bar> <foo@bar>` without putting it with quotes as `”<foo@bar>” <foo@bar>` when using word-encoding.

From what I understand of the RFC there is no legal way to represent decoded UTF-8 in an address. It would certainly be more correct if under these circumstances the “Name” would be at least put in quotes. At least the golang mail parser would parse that correctly then.

If this can’t be done then at least the [documentation](https://sendgrid.com/docs/for-developers/parsing-email/setting-up-the-inbound-parse-webhook/) should be updated because it’s currently very misleading (for `from` and `to`) :
> Email sender, as taken from the message headers.

Should be rephrased to something like:
> Email sender name and address. Both decoded into charset and separated by space. May not conform to a RFC 5322 e-mail address.

**Comment by yvesf** [comment]
tl;dr summary: The addresses in `from` from inbound parse can't be used to do something like send email reply to the sender. As alternative I can use the raw `headers` and parse the `From: ` in there.
The documentation is not helpful as it lacks the point that the email in `from` is useless except for displaying purposes.
